src/core/init/multithread/worker/track_choice_setter.ts

```diff
@@ -87,7 +87,7 @@ export default class TrackChoiceSetter {
       representations = new SharedReference<IRepresentationsChoice>(
         val.representations.getValue()
       );
-      ref.setValue(objectAssign({}, ref.getValue(), {
+      ref.setValue(objectAssign({}, val, {
```

Comment:
this was not an error, just typescript not aware that `ref.getValue` because we checked for `isNullOrUndefined(val)` previously